haystack/core/pipeline/pipeline.py

```diff
@@ -782,9 +782,21 @@ def run(self, word: str):
                         tags={
                             "haystack.component.name": name,
                             "haystack.component.type": comp.__class__.__name__,
-                            "haystack.component.inputs": {k: type(v).__name__ for k, v in last_inputs[name].items()},
+                            "haystack.component.input_types": {
+                                k: type(v).__name__ for k, v in last_inputs[name].items()
+                            },
+                            "haystack.component.input_spec": {
+                                key: {"type": value.type.__name__, "senders": value.senders}
+                                for key, value in comp.__haystack_input__._sockets_dict.items()
+                            },
+                            "haystack.component.output_spec": {
+                                key: {"type": value.type.__name__, "senders": value.receivers}
+                                for key, value in comp.__haystack_output__._sockets_dict.items()
+                            },
```

Comment:
added these as part of my debugging experiments yesterday. Quite useful information to piece together things